src/parser/Parser.cpp

```diff
@@ -1,122 +1,115 @@
         // TODO
-        default:
-            // Check for literal - if no literal, just return null
-            statement = parseLiteral();
-            if (!statement) break;
-            // Check for operator
-            switch(m_currentToken.type) {
-                case TokenType::Plus:
-                case TokenType::Minus:
-                case TokenType::Star:
-                case TokenType::Slash:
-                case TokenType::Ampersand:
-                case TokenType::Lesser:
-                case TokenType::LesserEqual:
-                case TokenType::Greater:
-                case TokenType::GreaterEqual:
-                case TokenType::Equal:
-                case TokenType::EqualEqual:
-                case TokenType::BangEqual:
-                case TokenType::And:
-                case TokenType::Or:
-                default:
-                    // Otherwise return literal
-                    break;
-            }
     }
-    getNextToken();
-    return std::move(statement);
-}
 
-std::unique_ptr<AST::Expr> Parser::parseLiteral() {
-    std::unique_ptr<AST::Expr> statement;
-    switch (m_currentToken.type) {
-        case TokenType::Integer:
-            statement = util::make_unique<AST::IntegerLit>(
-                    AST::IntegerLit(m_currentToken.literal));
-            break;
-        case TokenType::Float:
-            statement = util::make_unique<AST::FloatLit>(
-                    AST::FloatLit(m_currentToken.literal));
-            break;
-        case TokenType::String:
-            statement = util::make_unique<AST::StringLit>(
-                    AST::StringLit(m_currentToken.literal));
-            break;
-        case TokenType::True:
-        case TokenType::False:
-            statement = util::make_unique<AST::BooleanLit>(
-                    AST::BooleanLit(m_currentToken.literal));
-            break;
-        default:
-            // todo : exceptions, details
-            return std::unique_ptr<AST::Expr>(nullptr);
+    if (m_currenttoken.type == TokenType::Return) {
+        eattoken(TokenType::Return);
+        std::unique_ptr<AST::Expr> expr = parseExpr();
+        return std::unique_ptr<AST::ReturnStmt>(new AST::ReturnStmt(std::move(expr)));
+    }
+
+    // assignment or expr statement
+    std::unique_ptr<AST::Expr> expr = parseExpr();
+    if (m_currenttoken.type == TokenType::Equal) {
+        eattoken(TokenType::Equal);
+        std::unique_ptr<AST::Expr> rval = parseExpr();
+        // TODO: add checks that lval is valid.
+        return std::unique_ptr<AST::AssignStmt>(new AST::AssignStmt(std::move(expr), std::move(rval)));
+    } else {
+        return std::unique_ptr<AST::ExprStmt>(new AST::ExprStmt(std::move(expr)));
     }
-    getNextToken();
-    return std::move(statement);
-}
 
-std::unique_ptr<AST::FnDecl> Parser::parseFunction() {
-    return std::unique_ptr<AST::FnDecl>(nullptr);
+    // this is how I do error handling lol.
```

Comment:
I'm not sure if the other comment will show up but in case note, I assume we'll expand on error handling in the future?

In particular I do like the panic mode/synchronization method in Crafting Interpreters: http://www.craftinginterpreters.com/parsing-expressions.html#syntax-errors

Reply:
Yes, we should do synchronization, but we'll do it a level above this, and by just advancing until we see a semicolon.

Reply:
You mean a newline?